src/signals/incident/components/form/MapSelectors/Asset/Selector/WfsLayer/StatusLayer/utils.ts

```diff
@@ -0,0 +1,44 @@
+import type {
+  CheckedFeatureType,
+  Feature,
+  FeatureType,
+  ReportedFeatureType,
+} from 'signals/incident/components/form/MapSelectors/types'
+
+export const getIsReported = (
+  feature: Feature | undefined,
+  reportedFeatureType: ReportedFeatureType | undefined
+) => {
+  if (!feature || !reportedFeatureType) {
+    return false
+  }
+  return Boolean(
+    feature.properties[reportedFeatureType.isReportedField] ===
+      reportedFeatureType.isReportedValue
+  )
+}
+
+export const getIsChecked = (
+  feature: Feature | undefined,
+  checkedFeatureType: CheckedFeatureType | undefined
+): boolean => {
+  if (!feature || !checkedFeatureType) {
+    return false
+  }
+  const isCheckedValue = feature.properties[checkedFeatureType.isCheckedField]
+  return isCheckedValue
+    ? Boolean(
+        checkedFeatureType.isCheckedValues.includes(isCheckedValue.toString())
+      )
+    : false
+}
+
+export const getReportedFeatureType = (featureTypes: FeatureType[]) =>
+  featureTypes.find(
+    ({ typeValue }) => typeValue === 'reported'
+  ) as ReportedFeatureType
```

Comment:
Casting the return value to type `ReportedFeatureType` will provide false positives, since the result of a `find()` call can yield both a value of type `ReportedFeatureType` as well as `typeof undefined`.
Same goes for `getCheckedFeatureType`.